src/shared.js

```diff
@@ -532,19 +532,28 @@ export const checkDerivedVariables = async (participantObjToken) => {
 }
 
 export const updateBox = async (box) => {
-  logAPICallStartDev('updateBox');
-  const idToken = await getIdToken();
-  let requestObj = {
-      method: "POST",
-      headers:{
-          Authorization:"Bearer "+idToken,
-          "Content-Type": "application/json"
-      },
-      body: JSON.stringify(convertToFirestoreBox(box))
-  }
-  const response = await fetch(`${api}api=updateBox`, requestObj);
-  logAPICallEndDev('updateBox');
-  return response.json();
+    try {
+        const idToken = await getIdToken();
+        const requestObj = {
+            method: "POST",
+            headers:{
+                Authorization:"Bearer "+idToken,
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(convertToFirestoreBox(box)),
+        }
+
+        const response = await fetch(`${api}api=updateBox`, requestObj);
+        
+        if (!response.ok) {
+            throw new Error(`API responded with status: ${response.status}`);
+        }
+        
+        return await response.json();
+    } catch (error) {
+        console.error('Failed to update box:', error);
+        throw error;
```

Comment:
just confirming that calls to updateBox() will react appropriately if an error is returned? It looks like before we were always returning JSON objects.

Reply:
I believe it will. Here's the important part of `addEventAddSpecimensToListModalButton()`, where `updateBox is called:

```javascript
const boxToUpdate = prepareBoxToUpdate(currBoxId, boxList, boxIdAndBagsObj, locations, addedTubes);
            showAnimation();
            try {
                const boxUpdateResponse = await updateBox(boxToUpdate);
                hideAnimation();
                if (boxUpdateResponse.code === 200) {
                    updateShippingStateAddBagToBox(currBoxId, bagId, boxToUpdate, boxUpdateResponse.data);
                    await startShipping(appState.getState().userName, true, currBoxId);
                } else {
                    console.error('Failed to update box.', boxUpdateResponse);
                    showNotifications({ title: 'Error Adding Specimen(s)', body: `There was an error adding ${bagId.split(' ')[0]}. Please try again.` });
                }
            } catch (error) {
                hideAnimation();
                console.error('Failed to update box.', error);
                showNotifications({ title: 'Error Adding Specimen(s)', body: `There was an error adding ${bagId.split(' ')[0]}. Please try again.` });
            }
```
So a response other than 200 is handled in the `else {}` block and a thrown error is handled in the `catch {}` block.